### ISSUE TITLE
Battery capacity stuck at 100% if "last_full_capacity = true"

### ISSUE THREAD

**Issue by mbb78** [i3/i3status#169]
**This is a copy of the issue I wrongly filed in i3/i3 repository:**

Output of `i3 --moreversion 2>&- || i3 --version`:
```
Binary i3 version:  4.12 (2016-03-06, branch "4.12") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.12 (2016-03-06, branch "4.12") (pid 865)
Loaded i3 config: /home/mochobb/.i3/config (Last modified: sex 07 out 2016 18:16:35 WEST, 1293372 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

URL to a logfile as per http://i3wm.org/docs/debugging.html:

[logs.i3wm.org/logs/5651442522128384.bz2](logs.i3wm.org/logs/5651442522128384.bz2)
[i3bar.bar-0.log.txt](https://github.com/i3/i3/files/546006/i3bar.bar-0.log.txt)

**What I did:**

I changed i3status.conf to indicate the correct "uevent" path and set `last_full_capacity = true`

My current i3status.conf battery block:
```
battery 0 {
#	format = "%status %percentage %remaining %emptytime"
	format = " %status %percentage %remaining "
	format_down = " no battery "
	last_full_capacity = false
	integer_battery_capacity = true
        hide_seconds = true
#	status_chr = ""
	status_chr = "⚡"
#	status_bat = ""
#	status_bat = "bat"
	status_bat = "☉"
#	status_full = ""
	status_full = "☻"
        low_threshold = 10
  	threshold_type = percentage
        path = "/sys/class/power_supply/C11F/uevent"
}
```
My current "uvent" file:
```
POWER_SUPPLY_NAME=C11F
POWER_SUPPLY_STATUS=Full
POWER_SUPPLY_PRESENT=1
POWER_SUPPLY_TECHNOLOGY=Li-ion
POWER_SUPPLY_CYCLE_COUNT=0
POWER_SUPPLY_VOLTAGE_MIN_DESIGN=14800000
POWER_SUPPLY_VOLTAGE_NOW=16293000
POWER_SUPPLY_CURRENT_NOW=0
POWER_SUPPLY_CHARGE_FULL_DESIGN=1698000
POWER_SUPPLY_CHARGE_FULL=1698000
POWER_SUPPLY_CHARGE_NOW=1698000
POWER_SUPPLY_CAPACITY=100
POWER_SUPPLY_CAPACITY_LEVEL=Full
POWER_SUPPLY_MODEL_NAME=Primary
POWER_SUPPLY_MANUFACTURER= Hewlett-Packard 
POWER_SUPPLY_SERIAL_NUMBER=00 CC 10 A2 00 64 30 00
```
**What I saw:**

The battery current capacity is always shown as 100%, irrespective of the value directly got from the "uevent" file. This doesn't happen if I set `last_full_capacity = false`.

EDIT: if i3status.conf is edited setting `last_full_capacity = true` and i3 reloaded, this behaviour isn't visible. It only happens if one logs out and back in or after rebooting. Setting `last_full_capacity = false` consistently results in correct behaviour across reboots.

**What I expected instead:**

I expect the current capacity to be updated as the battery looses it's capacity when the power cord is unplugged. The correct behaviour occurs if I set `last_full_capacity = false`.

Thank you very much for taking the time to read this.

**Comment by stapelberg** [comment]
I think you’re running into the case documented in https://github.com/i3/i3status/blob/707ceffc8b556a3797651d78c0c0f37bae9884ea/src/print_battery_info.c#L508 (or a very similar case):

>     /* Some batteries report POWER_SUPPLY_CHARGE_NOW=<full_design> when fully
>      * charged, even though that’s plainly wrong. For people who chose to see
>      * the percentage calculated based on the last full capacity, we clamp the
>      * value to 100%, as that makes more sense.
>      * See http://bugs.debian.org/785398 */
>      if (last_full_capacity && batt_info.percentage_remaining > 100) {
>          batt_info.percentage_remaining = 100;
>      }

Can you please provide a uevent file when your battery is discharging, ideally at somewhere like 75%?

My current theory is that the uevent values your battery reports cannot be trusted.

**Comment by mbb78** [comment]
Hi!

Here's the current uvent at 73%:
```
POWER_SUPPLY_NAME=C11F
POWER_SUPPLY_STATUS=Discharging
POWER_SUPPLY_PRESENT=1
POWER_SUPPLY_TECHNOLOGY=Li-ion
POWER_SUPPLY_CYCLE_COUNT=0
POWER_SUPPLY_VOLTAGE_MIN_DESIGN=14800000
POWER_SUPPLY_VOLTAGE_NOW=15956000
POWER_SUPPLY_CURRENT_NOW=1090000
POWER_SUPPLY_CHARGE_FULL_DESIGN=1706000
POWER_SUPPLY_CHARGE_FULL=1706000
POWER_SUPPLY_CHARGE_NOW=1248000
POWER_SUPPLY_CAPACITY=73
POWER_SUPPLY_CAPACITY_LEVEL=Normal
POWER_SUPPLY_MODEL_NAME=Primary
POWER_SUPPLY_MANUFACTURER= Hewlett-Packard 
POWER_SUPPLY_SERIAL_NUMBER=00 CC 10 AA 00 4A 32 00
```
Thanks,

**Comment by eplanet** [comment]
I'm surprised that your `uevent` file provides different values for `POWER_SUPPLY_CHARGE_FULL_DESIGN` between the two examples given. However, if I use your `uevent` values to do some unit testing, I cannot reproduce any bug of any kind: the correct percentages are displayed.

Is your second `uevent` supposed to display 100%? Is the `%remaining` part correct by the way? Which i3status version are you running?

**Comment by mbb78** [comment]
Hi!

On the first example the power cord was plugged and the battery was fully charged. I just posted it so you could see its output. The second output had the power cord unplugged and the capacity was at 73%, because @stapelberg asked me to post it. If you're referring to the differences between the design capacities in both posts, then I can't explain them, neither I know if that's supposed to happen. I have it plugged and fully now, with the design and full capacities coincident with the 2nd post. Either way, I don't see how it could prevent the capacity from being updated.

I'm using i3status-manjaro v2.10.r56.g08ee3bd-1. Do you think I should try using i3status vanilla instead? I didn't even knew I was a manjaro customized package! Maybe I'll try to change and if it works I'll close this issue and report at the manjaro forum.

Thanks,

**Comment by Airblader** [comment]
AFAIK the manjaro package only provides a different default configuration. It doesn't actually change the source code, so let's keep the issue here.

**Comment by mbb78** [comment]
Ok! I was just trying to eliminate an hypotheses (which was unknown to me).

**Comment by eplanet** [comment]
@mbb78, I might still don't get what the problem is: your i3status is displaying 100% while the `uevent` values report 73%?

**Comment by mbb78** [comment]
No. The display is correct if `last_full_capacity = false` and gets stuck at 100% otherwise.

**Comment by eplanet** [comment]
Ok, so that is definitely an issue I cannot reproduce. If I take your second `uevent`, that has 73%, put the exact same i3status.conf as provided (whatever `last_full_capacity` value: false, true, unset), it displays 73%…

**Comment by mbb78** [comment]
It is consistently reproducible on my machine, but not noticeable if I edit the .conf file and reload i3. It just becomes visible after login/logout or reboot (as some of other i3 settings). I did search for other similar reports on the web but I haven't found any. One user has reported this issue at manjaro forum thought, and changing `last_full_capacity` to false, then rebooting the system, solved it. So this might be rare, but I'm not the only one.

**Comment by stapelberg** [comment]
Try testing by just running “i3status” in a terminal window, in order to isolate any strange effects coming into your test setup because of i3/i3bar.

**Comment by mbb78** [comment]
Just did it at 64%. I first ran `i3status` whit my current config (`last_full_capacity = false`) and the output was correct. Then, I exited i3status, edited the .conf file to set `last_full_capacity = true` and ran `i3status` again: the output showed 100% (the remaining time is coherent though - it always behaved like this).
